src/compiler/checker.ts

```diff
@@ -11975,6 +11975,10 @@ namespace ts {
             // Variables declared with 'const'
             // Get accessors without matching set accessors
             // Enum members
+            // Unions and intersections of the above
+            if (symbol.flags & SymbolFlags.SyntheticProperty) {
+                return forEach(symbol.declarations, decl => isReadonlySymbol(getSymbolOfNode(decl)));
+            }

```

Comment:
This could probably be more efficient. When a symbol has multiple declarations (such as an overloaded method) we'll end up doing the same check multiple times even though the outcome is always false. Ideally we'd have a `SymbolFlags.Readonly` that we compute in the same way as `SymbolFlags.Optional`, but we're out of flag bits. Perhaps it would be better to compute the readonly state in `createUnionOrIntersectionProperty` (but calling `isReadonlySymbol` for each underlying symbol) and then record the result in an `isReadonly` property on the symbol itself.